machine/v1beta1/types_vsphereprovider.go

```diff
@@ -198,8 +198,30 @@ type VSphereDisk struct {
 	// +kubebuilder:validation:Maximum=16384
 	// +required
 	SizeGiB int32 `json:"sizeGiB"`
+	// provisioningMode specifies the provisioning type to be used by this vSphere data disk.
+	// If not set, the setting will be provided by the default storage policy.
+	// +optional
+	ProvisioningMode ProvisioningMode `json:"provisioningMode,omitempty"`
 }
 
+// provisioningMode represents the various provisioning types available to a VMs disk.
+// +kubebuilder:validation:Enum=Thin;Thick;EagerlyZeroed
+type ProvisioningMode string
+
+var (
+	// ThinProvisioningMode creates the disk using thin provisioning. This means a sparse (allocate on demand)
+	// format with additional space optimizations.
+	ThinProvisioningMode ProvisioningMode = "Thin"
```

Comment:
The most common pattern I've seen for naming of constants for a type alias is `{TypeAlias}{ConstantName}`. In this case it would be:
```suggestion
	ProvisioningModeThin ProvisioningMode = "Thin"
```

I would encourage consistency with how this is commonly done throughout the various OpenShift repos.

Reply:
I can make this change, but will deviate from upstream CAPV.  The actual values will still match, just slight var difference.  (cc: @JoelSpeed).

Reply:
I agree with @everettraven on this one to be consistent with OpenShift APIs, when it comes to converting to upstream, this won't really make a difference to us

Reply:
Sounds good.  I'll keep this in mind.